### Added

pdrflp0 opened the pull request:
- Method `getCategoriesByLine(String lineName)` in `CategoryDAO` to retrieve all categories associated with a specific line.
   - Method `getModelsByCategory(CategoryEntity category)` in `ModelDAO` to retrieve all models associated with a specific category.
   - Logic in the `ScreenController` class to display corresponding categories and models when a line is selected.
   - Logic to handle line selection and display corresponding categories and models in the user interface.
   - Necessary imports for the added classes and their respective packages.

### Changed
   - Method `handleLineSelection()` in the `ScreenController` class was modified to call the `displayModels(selectedLine)` and `expandModels()` methods after a line is selected.
   - Method `displayModels(String selectedLine)` in the `ScreenController` class was modified to retrieve categories associated with the selected line and display models corresponding to those categories.
   - Method `expandModels()` in the `ScreenController` class was modified to expand the models pane when categories and models are displayed.

pdrflp0 committed:
### Added
   - Method `getCategoriesByLine(String lineName)` in `CategoryDAO` to retrieve all categories associated with a specific line.
   - Method `getModelsByCategory(CategoryEntity category)` in `ModelDAO` to retrieve all models associated with a specific category.
   - Logic in the `ScreenController` class to display corresponding categories and models when a line is selected.
   - Logic to handle line selection and display corresponding categories and models in the user interface.
   - Necessary imports for the added classes and their respective packages.

### Changed
   - Method `handleLineSelection()` in the `ScreenController` class was modified to call the `displayModels(selectedLine)` and `expandModels()` methods after a line is selected.
   - Method `displayModels(String selectedLine)` in the `ScreenController` class was modified to retrieve categories associated with the selected line and display models corresponding to those categories.
   - Method `expandModels()` in the `ScreenController` class was modified to expand the models pane when categories and models are displayed.

Bacuriim commented on src/main/java/br/eletra/frontend/controllers/ScreenController.java:
```diff
@@ -20,43 +31,73 @@ public class ScreenController {
-                    subcategoryItem.getChildren().add(new TreeItem<>(model));
+        try {
+            LineEntity line = lineDAO.getLineByName(selectedLine);
+            if (line != null) {
```
Essa verificação é inútil, visto que ela já ocorreu na linha 66, por favor remover.

pdrflp0 replied:
Pendência resolvida! ✅

Bacuriim reviewed (changes requested):
Foi encontrada apenas uma pendência, se atentar mais as verificações (IF-ELSES) no código para não gerar dupla verificação ou redundâncias.

pdrflp0 committed:
### Added
   - BD.sql file with all the codes used to create the tables.

### Removed
   - Unnecessary "IF" on line 77 of the `ScreenController` requested in the PR.

pdrflp0 committed:
Rename bd.sql.txt to bd.sql
